### DISCUSSION TITLE
Return value of `rename` in vfile-rename

### DISCUSSION THREAD

**Discussion by webpro** [vfile/vfile#75] [Q&A]
The `vfile-rename` function works fine, but it was pretty subtle to me that the provided `file` was not renamed:

```
const vFile = readSync(source);
rename(vFile, {});
```

However, the returned `file` does include the rename and has the full `history`:

```
vFile = rename(vFile, {});
```

Is this intentional? I'd think a rename would modify the passed file. Alternatively, I guess this should be documented more clearly.

**Comment by webpro** [comment]
Never mind, the implementation is fine, looks like no surprises here (https://github.com/vfile/vfile-rename/blob/main/index.js).

There is something very odd going on with my TypeScript setup. Sorry, feel free to close/delete this.

**Reply by wooorm** [reply]
perhaps you have a version mismatch? `readSync` returns an instance of an old VFile class?

**Reply by wooorm** [reply]
That’s my guess, that https://github.com/vfile/vfile-rename/blob/103666cc3a5475df5e9382cdc209f17d1caead41/index.js#L44 is creating a new file

**Comment by webpro** [comment]
Ah yes, that was probably it. Although I'm sure had the latest & greatest of all dependencies, I guess both `npm link` and `npm install --legacy-peer-deps` may eventually result in mismatched versions somehow. Obviously I can't reproduce it anymore... Leaving the safety net here to avoid potential annoyances: https://github.com/webpro/markdown-rambler/blob/264f7b952098ae297fd97f7900fa911a91eddf4e/src/MarkdownRambler.ts#L185

Thanks for the swift response!

**Reply by wooorm** [reply]
Ohh yeah linking unfortunately does that, I’ve had similar problems before. It’s one part of what I don’t like about TypeScript (which we now use everywhere). It forces the use of `instanceof`, which breaks across realms and classes, compared to the more resilient ducktyping that was in unified before :(
